src/storage/mvcc/txn.rs

```diff
@@ -199,14 +179,34 @@ impl<S: Snapshot> MvccTxn<S> {
         Ok(())
     }
 
+    fn check_non_pessimistic_lock_conflict(&self, for_update_ts: u64, lock: &Lock) -> Result<()> {
+        // If request's for_update_ts is greater than lock's for_update_ts,
+        // we can overwrite it because isolation is guaranteed by pessimistic
+        // locks and no other transaction can read the old data because the lock
+        // is replaced atomicly.
+        //
+        // Optimistic lock's for_update_ts is its start_ts.
+        let lock_for_update_ts = if lock.for_update_ts > 0 {
```

Comment:
When will it happen? Here we meet a pessimistic lock which does not belong to current transaction when we prewrite, which I think this shouldn't happen.

Reply:
It is a non-pessimistic lock. The google doc has more info.